File: totosuki/ABC/B/304.py
import math
N = int(input())

if N < 10 ** 3:
  print(N)
elif N < 10 ** 4:
  print(math.floor(N / 10) * 10)
elif N < 10 ** 5:
  print(math.floor(N / 100) * 100)
elif N < 10 ** 6:
  print(math.floor(N / 1000) * 1000)
elif N < 10 ** 7:
  print(math.floor(N / 10000) * 10000)
elif N < 10 ** 8:
  print(math.floor(N / 100000) * 100000)
else:
  print(math.floor(N / 1000000) * 1000000)

File: OCH_Tao/ABC/A/304.js
const input = (require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n"));
const N = parseInt(input.shift(), 10);
let X = [];
let Y = [];
let cnt;
for (const i of input) {
  const [A, B] = i.split(" ");
  X.push(A);
  Y.push(parseInt(B, 10));
  cnt = cnt === undefined ? parseInt(B) : Math.min(cnt, parseInt(B, 10));
}
let j = Y.indexOf(cnt);
console.log(X.slice(j).concat(X.slice(0, j)).join("\n"));